core/object/worker_thread_pool.cpp

```diff
@@ -60,6 +61,14 @@ void WorkerThreadPool::_process_task(Task *p_task) {
 		set_current_thread_safe_for_nodes(false);
 		pool_thread_index = thread_ids[Thread::get_caller_id()];
 		ThreadData &curr_thread = threads[pool_thread_index];
+		// Since the WorkerThreadPool is started before the script server,
+		// it's pre-created threads can't have ScriptServer::thread_enter() called on them early.
```

Comment:
```suggestion
		// its pre-created threads can't have ScriptServer::thread_enter() called on them early.
```

I can only say that all this makes sense on the spiritual level, but can't really review it. So here's a grammar nitpick instead :)